packages/backend/src/services/SavedChartsService/SavedChartService.ts

```diff
@@ -148,4 +148,30 @@ export class SavedChartService {
         });
         return newSavedChart;
     }
+
+    async duplicate(
+        user: SessionUser,
+        projectUuid: string,
+        chartUuid: string,
+    ): Promise<SavedChart> {
+        if (user.ability.cannot('create', 'SavedChart')) {
+            throw new ForbiddenError();
+        }
+        const chart = await this.savedChartModel.get(chartUuid);
+        const duplicatedChart = {
+            ...chart,
+            name: `Copy of ${chart.name}`,
+        };
+        const newSavedChart = await this.savedChartModel.create(
+            projectUuid,
+            duplicatedChart,
+        );
+        analytics.track({
```

Comment:
@TuringLovesDeathMetal is this better as another create event but with a property?

```
            event: 'saved_chart.created',
             userId: user.userUuid,
             properties: {
                 duplicated: true,
                 ...SavedChartService.getCreateEventProperties(newSavedChart)
             },
               
         });
 ```

Reply:
Reason if you wanted to sum chart creation events  you probably want to include duplications

Reply:
yes please! 

Is there some way to have two events:

- one event for `duplicated_chart_created`
- one event for `saved_chart_created`

Reply:
Could we also please add `saved_query_id` to the `duplicated_chart_created`

ideally something like:
`duplicate_of_saved_query_id` for the chart_id it's being created from and `new_saved_query_id` for the new chart being created